File: python/algo1806.py
n, s = map(int , input().split())

seq = list(map(int , input().split()))

top = 0 ; bottom = 0
sum = 0
ans = 100001

while top <= bottom and bottom < n:
    if sum < s:
        sum += seq[bottom]
        bottom += 1

    while sum >= s:
        ans = min(ans, bottom - top)
        sum -= seq[top]
        top += 1


if ans >= 100001: print(0)
else: print(ans)

File: c/algo1806.c
#include <stdio.h>

int sum(int* arr, int start, int end) {
    int sum = 0;

    for (int i = start ; i < end ; i++){
        sum += arr[i];
    }

    return sum;
}

int main() {
    int n, s;
    int top = 0, bottom = 0;
    int seq[100000];
    int ans = 100000;

    scanf("%d %d", &n, &s);

    for (int i = 0 ; i < n ; i++) {
        scanf("%d", &seq[i]);
    }

    while (top <= bottom && bottom <= n) {
        if (sum(seq, top, bottom) < s) {
            bottom++;
        } else {
            ans = (ans > bottom - top) ? bottom - top : ans;
            top += 1;
        }
    }  

    printf("%d\n", ans);
    return 0;
}